R/toolISOhistorical.R

```diff
@@ -111,61 +112,293 @@ toolISOhistorical <- function(m, mapping = NULL, additional_mapping = NULL, over
+        } else {
+          if (!all(a$toISO %in% getItems(additional_weight, dim = 1))) {
+            stop(paste0(
+              "Invalid additional weight, missing countries: ",
+              paste0(setdiff(a$toISO, getItems(additional_weight, dim = 1)), collapse = ", ")
+            ))
+          }
+          weight <- additional_weight[a$toISO, , ]
         }
+        
+        mTr <- toolAggregate(m[a$fromISO, subTime, ],
+                             mapping[is.element(mapping$toISO, a$toISO), c("fromISO", "toISO")], weight = weight,
+                             negative_weight = "allow")
+        ## aggregation of countries
       } else {
-        if (!all(a$toISO %in% getItems(additional_weight, dim = 1))) {
-          stop(paste0(
-            "Invalid additional weight, missing countries: ",
-            paste0(setdiff(a$toISO, getItems(additional_weight, dim = 1)), collapse = ", ")
-          ))
-        }
-        weight <- additional_weight[a$toISO, , ]
+        mTr <- toolAggregate(m[a$fromISO, subTime, ],
+                             mapping[is.element(mapping$toISO, a$toISO), c("fromISO", "toISO")], weight = NULL)
+      }
+      # fill data
+      if (overwrite == TRUE) {
+        m[a$toISO, subTime, ] <- mTr
+      } else {
+        m[a$toISO, subTime, ][is.na(m[a$toISO, subTime, ])] <- mTr[is.na(m[a$toISO, subTime, ])]
+      }
+    } # a in tr - transitions
+    
+    # delete old lines
+    for (b in mapping$fromISO) {
+      if (is.element(b, getItems(m, dim = 1.1))) {
+        m <- m[-which(getItems(m, dim = 1.1) == b), , ]
       }
-
-      mTr <- toolAggregate(m[a$fromISO, subTime, ],
-                           mapping[is.element(mapping$toISO, a$toISO), c("fromISO", "toISO")], weight = weight,
-                           negative_weight = "allow")
-      ## aggregation of countries
-    } else {
-      mTr <- toolAggregate(m[a$fromISO, subTime, ],
-                           mapping[is.element(mapping$toISO, a$toISO), c("fromISO", "toISO")], weight = NULL)
-    }
-    # fill data
-    if (overwrite == TRUE) {
-      m[a$toISO, subTime, ] <- mTr
-    } else {
-      m[a$toISO, subTime, ][is.na(m[a$toISO, subTime, ])] <- mTr[is.na(m[a$toISO, subTime, ])]
     }
-  } # a in tr - transitions
-
-  # delete old lines
-  for (b in mapping$fromISO) {
-    if (is.element(b, getItems(m, dim = 1.1))) {
-      m <- m[-which(getItems(m, dim = 1.1) == b), , ]
+  } else if (ndim(m, dim = 1) == 2) {
+    # bilateral process needs to iterate over dim 1 and dim 2 due to potential presence of different transitions
+    
+    origDim1 <- getSets(m)[1]
+    origDim2 <- getSets(m)[2] ## store original set names for later renaming, but for now rename for easy list calls later
+    
+    getSets(m)[1] <- "dim11"
+    getSets(m)[2] <- "dim12"
+    
+    
+    # loop over all transitions
+    for (a in tr) {
+      # check if new regions of transition exists in m
+      toISOmissing <- !all(is.element(a$toISO, getItems(m, dim = 1.1)))
```

Comment:
You could consider this:
```suggestion
      toISOmissing <- !all(a$toISO %in% getItems(m, dim = 1.1))
```
I find it a bit nicer, but that's personal preference of course.